### PR TITLE
[WNMGDS-2505] Fix react-aria's react support and remove downshift

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

WNMGDS-2505

When you try to use the keyboard to navigate through options in the Autocomplete, it won't work in Preact. I put in a pull request to Adobe Spectrum to fix it, but unfortunately that fix creates other problems in their libraries. The Adobe Spectrum team has catalogued many issues with Preact, and they can't commit to supporting it officially. Because I need to fix this and release it, I've copied their library into our repo so I could make the fix there. Take a look at 01e01e2ac37845ad82903b137218287f9ebcdc7f to see the fix (in 3fb11774c5ffc46a2b8f97338ff4098df9917a66 I added a comment).

- Added the `react-aria` and `react-stately` source and fixed the Preact bug
- Removed `downshift` and all references to it

## How to test

1. `yarn storybook:preact` and test the Autocomplete with the keyboard

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### CODE REVIEW

File: packages/design-system/src/components/Dropdown/DropdownMenuSection.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { DropdownMenuOption } from './DropdownMenuOption';
-import { ListState, Node } from 'react-stately';
-import { useListBoxSection } from 'react-aria';
+import { ListState, Node } from '../react-aria';
+import { useListBoxSection } from '../react-aria';
```
Comment (zarahzachz) [inline packages/design-system/src/components/Dropdown/DropdownMenuSection.tsx:4]: could these lines be combined now that they're imported from the same place?

File: packages/design-system/src/components/Dropdown/DropdownMenuOption.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React, { useRef } from 'react';
-import { ListState, Node } from 'react-stately';
+import { ListState, Node } from '../react-aria';
 import { SvgIcon } from '../Icons';
-import { useOption } from 'react-aria';
+import { useOption } from '../react-aria';
```
Comment (zarahzachz) [inline packages/design-system/src/components/Dropdown/DropdownMenuOption.tsx:4]: could these lines be combined now that they're imported from the same place?

File: packages/design-system/src/components/Dropdown/DropdownMenu.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import React, { RefObject, useRef } from 'react';
 import { DropdownMenuOption } from './DropdownMenuOption';
 import { DropdownMenuSection } from './DropdownMenuSection';
-import { ListState, OverlayTriggerState } from 'react-stately';
-import { AriaPopoverProps, AriaListBoxOptions, useListBox } from 'react-aria';
+import { ListState, OverlayTriggerState } from '../react-aria';
+import { AriaPopoverProps, AriaListBoxOptions, useListBox } from '../react-aria';
```
Comment (zarahzachz) [inline packages/design-system/src/components/Dropdown/DropdownMenu.tsx:5]: could these lines be combined now that they're imported from the same place?

File: packages/design-system/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -5,8 +5,8 @@ import useAutofocus from '../utilities/useAutoFocus';
 import { FormFieldProps, FormLabel, useFormLabel } from '../FormLabel';
 import { SvgIcon } from '../Icons';
 import { getFirstOptionValue, isOptGroup, parseChildren, validateProps } from './utils';
-import { Item, Section, useSelectState } from 'react-stately';
-import { HiddenSelect, useButton, useSelect } from 'react-aria';
+import { Item, Section, useSelectState } from '../react-aria';
+import { HiddenSelect, useButton, useSelect } from '../react-aria';
```
Comment (zarahzachz) [inline packages/design-system/src/components/Dropdown/Dropdown.tsx:9]: could these lines be combined now that they're imported from the same place?

File: package.json [hunk 5]
```diff
@@ -12,8 +12,8 @@
     "build:healthcare": "yarn gulp build --package packages/ds-healthcare-gov",
     "build:medicare": "yarn gulp build --package packages/ds-medicare-gov",
     "build:cmsgov": "yarn gulp build --package packages/ds-cms-gov",
-    "build:storybook": "./scripts/build-storybook.sh",
-    "build:storybook:docs": "./scripts/build-storybook.sh docs",
+    "build:storybook": "PREACT=true ./scripts/build-storybook.sh",
```
[the remark this answers is not in the document]
Comment (pwolfert) [reply package.json:15]: That's probably a good idea. I wanted the default to be Preact again, but we do seem to have some outstanding issues (like with Tooltip)

File: packages/design-system/src/components/Dropdown/DropdownMenuSection.tsx (hunk 1 above)
Comment (pwolfert) [reply packages/design-system/src/components/Dropdown/DropdownMenuSection.tsx:4]: Yeah, good call

**Review by zarahzachz — APPROVED** [review]

**Pull request merged**